File: examples/ruby/pagination/sandbox.rb
# frozen_string_literal: true

require 'daytona'

daytona = Daytona::Daytona.new

daytona.list(Daytona::ListSandboxesQuery.new(
               limit: 10,
               labels: { 'env' => 'dev' },
               states: [Daytona::SandboxState::STARTED],
               sort: Daytona::SandboxListSortField::CREATED_AT,
               order: Daytona::SandboxListSortDirection::DESC
             )).each do |sandbox|
  puts sandbox.id
end

File: examples/python/pagination/sandbox.py
from daytona import Daytona, ListSandboxesQuery, SandboxListSortDirection, SandboxListSortField, SandboxState


def main():
    daytona = Daytona()

    for sandbox in daytona.list(
        ListSandboxesQuery(
            limit=10,
            labels={"env": "dev"},
            states=[SandboxState.STARTED],
            sort=SandboxListSortField.CREATEDAT,
            order=SandboxListSortDirection.DESC,
        )
    ):
        print(sandbox.id)


if __name__ == "__main__":
    main()
